### PR TITLE
docs: rework README

### PR DESCRIPTION
**Pull request by lucab**
This moves docpages around and re-arranges the README so that it
provides a direct high-level overview of the project, putting all
the finer details inside sub-pages.

---

[Rendered](https://github.com/lucab/afterburn/blob/401fb2a58e4c60afb260a9072eb6fd88c22069c7/README.md)

### CODE REVIEW

**Comment by lucab** [comment]
@miabbott @bgilbert I think this arrangement covers the "quick overview" better and puts further details behind a level of indirection. How does it feel to you?

File: docs/contributing.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ contribution. See the [DCO](DCO) file for details.
 
 Due to their public nature, GitHub and mailing lists are not appropriate places for reporting vulnerabilities. Please refer to CoreOS's [security disclosure][disclosure] process when reporting issues that may be security related.
 
+[disclosure]: https://coreos.com/security/disclosure/
```
Comment (miabbott) [inline docs/contributing.md]: Should this be updated to point to Red Hat disclosure docs?

**Comment by miabbott** [comment]
@lucab love the reorganization of the docs; feels much more natural for someone discovering the project for the first time.

Thanks for your continued focus on improving our docs story!

**Comment by lucab** [comment]
@miabbott thanks, mind stamping it too then?

File: README.md [hunk 2]
```diff
@@ -4,140 +4,64 @@
...
-      - AFTERBURN_PACKET_IPV4_PUBLIC_GATEWAY_0
-      - AFTERBURN_PACKET_IPV4_PRIVATE_0
-      - AFTERBURN_PACKET_IPV4_PRIVATE_GATEWAY_0
-      - AFTERBURN_PACKET_IPV6_PUBLIC_0
-      - AFTERBURN_PACKET_IPV6_PUBLIC_GATEWAY_0
-  - vagrant-virtualbox
-    - Attributes
-      - AFTERBURN_VAGRANT_VIRTUALBOX_PRIVATE_IPV4
-      - AFTERBURN_VAGRANT_VIRTUALBOX_HOSTNAME
+Depending on the specific platform, the following services may run in the [initramfs](./dracut/30afterburn/) on first-boot:
```
Comment (bgilbert) [inline README.md]: `first boot`

File: README.md [hunk 3]
```diff
@@ -4,140 +4,64 @@
...
+ * setting local hostname
+ * injecting [network command-line arguments](./docs/usage/initrd-network-cmdline.md)
 
-Additionally, some attribute names are reserved for usage by [custom metadata providers][custom-metadata].
-These can be safely used by external providers on a platform not supported by Afterburn:
+The following features are conditionally available on some platforms as [systemd service units](./systemd/):
+ * installing public SSH keys for local system users
+ * retrieving [attributes](./docs/usage/attributes.md) from instance metadata
+ * checking-in to the provider in order to report a successful boot or instance provisioning
```
Comment (bgilbert) [inline README.md]: `checking in`

**Review by bgilbert — APPROVED** [review]
LGTM!

File: docs/contributing.md (hunk 1 above)
Comment (bgilbert) [reply docs/contributing.md]: Probably.  The CoreOS disclosure process is going away.

Comment (lucab) [reply docs/contributing.md]: Ah, good catch. I think I can drop this section at all, the repository already has the "security notice page" set and that is properly pointing toward Red Hat.

**Review by bgilbert — APPROVED** [review]

**Review by miabbott — APPROVED** [review]

**Comment by lucab** [comment]
I gave up on trying to convince Travis to report back on this. Doc changes only, manually merging.

**Pull request merged**